deploy/operator/statefulset.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
I assume this is changed in of of the next commits so it's kind of hard to review multiple times. I see why you did that as you are keeping the changes separate but as we are essentially redesigning how it works, would you mind just reducing those 10 commits to say about 3 (moving the manifests, generated changes+deletions of the old generated files, deps)?